tests/e2e/cleanup.sh

```
@@ -0,0 +1,37 @@
+    section "AGR Cleanup Tests"
+    echo "Test directory: $TEST_DIR"
+fi
+
+# Ensure we have at least one recording for cleanup to find
+if [ ! -d "$HOME/recorded_agent_sessions" ] || [ -z "$(ls -A "$HOME/recorded_agent_sessions" 2>/dev/null)" ]; then
+    # Create a recording first
+    $AGR record echo -- "cleanup test recording" </dev/null
+fi
+
+# Count current sessions
+SESSION_COUNT=$(find "$HOME/recorded_agent_sessions" -name "*.cast" 2>/dev/null | wc -l | tr -d ' ')
```

Shellcheck (0.11.0)

[warning] SESSION_COUNT appears unused. Verify use (or export if used externally).

(SC2034)